index.js

```diff
@@ -0,0 +1,241 @@
+//import libraries
+const dotenv = require('dotenv').config();
+const phoenix = require('@phoenixlan/phoenix.js');
+const amqp = require('amqplib/callback_api');
+const {
+    UTCDate
+} = require("@date-fns/utc");
+const {
+    addHours,
+    subMonths,
+    differenceInMilliseconds,
+    differenceInHours
+} = require("date-fns");
+const {
+    Client,
+    GatewayIntentBits,
+    EmbedBuilder,
+    Guild
+} = require('discord.js');
+//initialise the api
+phoenix.init("https://api.test.phoenixlan.no");
+//change this to token from login in the api
+phoenix.User.Oauth.setAuthState("", "");
+//variable that controls the months before the next event crew roles should be removed
+const timeBeforeNextEventRemove = 2;
+//change this, guild id to ensure proper guild selection
+let phoenixGuildId = "";
+//set the discord token variable with the token from .env file
+const DISCORD_TOKEN = process.env.BOT_TOKEN;
+//change this to the rabbitmq address and port
+amqp.connect('amqp://', function (error0, connection) {
+    if (error0) {
+        throw error0;
+    }
+    connection.createChannel(function (error1, channel) {
+        if (error1) {
+            throw error1;
+        }
+        //change this
+        let queue = '';
+
+
+        channel.assertQueue(queue, {
+            durable: true,
+        });
+
+        channel.consume(queue, async function (msg) {
+            console.log(msg.content.toString());
+            removeAllRoles();
+            updateRoles();
+        }, {
+            noAck: true,
+        });
+
+    });
+});
+const phoenixClient = new Client({
+    intents: [
+        GatewayIntentBits.Guilds,
+        GatewayIntentBits.GuildMessages,
+        GatewayIntentBits.MessageContent,
+        GatewayIntentBits.GuildMembers,
+    ],
+});
+
+async function handleRoleRemoval() {
+    const now = new UTCDate();
+    const events = await phoenix.getEvents();
+    const nextEventTimeOffsetted = events
+        .map(event => subMonths(new UTCDate(event["start_time"] * 1000), timeBeforeNextEventRemove))
+        .find(time => time.getTime() > now.getTime());
+    if (!nextEventTimeOffsetted || differenceInHours(nextEventTimeOffsetted, now) > 25) {
+        console.info("Skipping scheduled role removal");
+        return;
+    }
+
+    console.log("Removing roles");
+    removeAllRoles();
+}
+setInterval(() => {
+    handleRoleRemoval();
+}, 8.64e+7); // 1 day
+
+
+async function removeAllRoles() {
+    const guild = phoenixClient.guilds.cache.get(phoenixGuildId);
+    try {
+        let crews = await Promise.all((await phoenix.Crew.getCrews()));
```

Comment:
There's no need to make this mutable, as it is never changed. Furthermore, `Promise.all` is redundant here since you're working with a single promise, and it's already await'd.
```suggestion
        const crews = await phoenix.Crew.getCrews();
```

Reply:
DU LA TIL PROMISE.ALL